nselib/tls.lua

```diff
@@ -1340,6 +1340,12 @@ function client_hello(t)
   -- Set the session ID.
   table.insert(b, '\0')
 
+  if t.host and t.host.registry and t.host.registry.supported_ciphers then
+    DEFAULT_CIPHERS = {}
```

Comment:
You shouldn't be modifying `DEFAULT_CIPHERS`, since it's the default for all connections that we don't have prior info about. Instead, have a local var to point to the best option, like `local default_ciphers = t.host.registry.supported_ciphers or DEFAULT_CIPHERS`.